allium/lib/Parser/Parser.cpp

```diff
@@ -245,7 +245,6 @@ ParserResult<EffectCtorRef> Parser::parseEffectCtorRef() {
     Token identifier = lexer.take_next();
     if(identifier.type != Token::Type::identifier) {
         errors.push_back(SyntaxError("Expected identifier after \"do\".", identifier.location));
-        lexer.rewind(identifier);
```

Comment:
I believe the `do (<args with errors>)` case would have been handled strangely before the rewind was removed. Since the rewind brings the lexer back to the `do` token, the left paren wouldn't be seen, and the next error after `Expected identifier after "do"` would be `Expected a "<-" after the head of an effect implication.` (assuming that parseEffectCtorRef() was called from parseEffectImplication()). In general, I was trying to get rid of rewinds after errors in my previous PR but didn't catch this particular instance.